allennlp/semparse/executors/wikitables_variable_free_executor.py

```diff
@@ -0,0 +1,647 @@
+from typing import List, Dict, Tuple, Union
+from collections import defaultdict
+import logging
+
+from overrides import overrides
+
+from allennlp.semparse.worlds.world import ExecutionError
+from allennlp.semparse.contexts.table_question_knowledge_graph import MONTH_NUMBERS
+from allennlp.semparse.executors.executor import Executor, NestedList
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class Date:
+    def __init__(self, year: int, month: int, day: int) -> None:
+        self.year = year
+        self.month = month
+        self.day = day
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, Date):
+            return False
+        year_is_same = self.year == -1 or other.year == -1 or self.year == other.year
+        month_is_same = self.month == -1 or other.month == -1 or self.month == other.month
+        day_is_same = self.day == -1 or other.day == -1 or self.day == other.day
+        return year_is_same and month_is_same and day_is_same
+
+    def __gt__(self, other) -> bool:
+        # pylint: disable=too-many-return-statements
+        if not isinstance(other, Date):
+            return False
+        if self.year == -1 or other.year == -1:
+            return False
+        if self.year != other.year:
+            return self.year > other.year
+        # The years are equal and not -1
+        if self.month == -1 or other.month == -1:
+            return False
+        if self.month != other.month:
+            return self.month > other.month
+        # The months and years are equal and not -1
+        if self.day == -1 or other.day == -1:
+            return False
+        return self.day > other.day
+
+    def __ge__(self, other) -> bool:
+        if not isinstance(other, Date):
+            return False
+        return self > other or self == other
+
+
+
+class WikiTablesVariableFreeExecutor(Executor):
+    def __init__(self, table_data: List[Dict[str, str]]) -> None:
+        super().__init__(table_data)
+        self._table_data = table_data
+
+    @overrides
+    def _handle_expression(self, expression_list: NestedList):
+        if isinstance(expression_list, list) and len(expression_list) == 1:
+            expression = expression_list[0]
+        else:
+            expression = expression_list
+        if isinstance(expression, list):
+            # This is a function application.
+            function_name = expression[0]
+        else:
+            # This is a constant (like "all_rows" or "2005")
+            return self._handle_constant(str(expression))
+        try:
+            function = getattr(self, f"_{function_name}")
```

Comment:
I think we made these methods private previously because they were combined with the `World`.  I'd recommend just making the language predicates public, and removing the underscore here.  That's also a clear way to separate the helper functions from the language functions.

Reply:
Done.